socrata.module

```diff
@@ -108,84 +108,23 @@ function _socrata_dbg($msg, $vars = array()) {
 }
 
 /**
- * Util function to resolve redirects.
- */
-function _socrata_resolve_redirects($endpoint) {
-  // Make sure we have a valid-looking endpoint.
-  if (substr_compare($endpoint, '.json', -5)) {
-    $endpoint .= '.json';
-  }
-
-  // Create a new cURL resource.
-  $ch = curl_init();
-  if ($ch) {
-    // Pull and set custom curl options.
-    // @FIXME
-// Could not extract the default value because it is either indeterminate, or
-// not scalar. You'll need to provide a default value in
-// config/install/socrata.settings.yml and config/schema/socrata.schema.yml.
-$curlopts = \Drupal::config('socrata.settings')->get('socrata_curl_options');
-
-// Need to look at hook_socrata_curl_options_alter in D7 module.
-//    \Drupal::moduleHandler()->alter('socrata_curl_options', $curlopts);
-    curl_setopt_array($ch, $curlopts);
-
-    // Set required curl options.
-    $req_curlopts = array(
-      CURLOPT_HEADER => TRUE,
-      CURLOPT_RETURNTRANSFER => TRUE,
-    );
-    curl_setopt_array($ch, $req_curlopts);
-
-    // Execute the request and cleanup.
-    curl_setopt_array($ch, $curlopts);
-    do {
-      $url = Url::fromUri($endpoint, array('query' => array('$limit' => 1), 'absolute' => TRUE))->toString();
-
-      curl_setopt($ch, CURLOPT_URL, $soda_url);
-      $resp = curl_exec($ch);
-      if (FALSE !== $resp) {
-        // Pull info from response and see if we had an error.
-        $info = curl_getinfo($ch);
-        if ($info['http_code'] >= 400) {
-          break;
-        }
-        else {
-          // Split out response headers into name => value array.
-          list($headers, $data) = explode("\r\n\r\n", $resp, 2);
-          $headers = _socrata_parse_headers($headers);
-
-          // Test for redirect in the event curl wasn't able
-          // to automagically follow due to server config.
-          if (!empty($headers['location'])) {
-            $endpoint = $headers['location'];
-          }
-        }
-      }
-    } while (FALSE !== $resp && !empty($headers['location']));
-  }
-
-  return $endpoint;
-}
-
-/**
  * Util function to parse out HTTP response headers.
  */
 function _socrata_parse_headers($headers) {
```

Comment:
Since _socrata_parse_headers() is only called by execute() in the SocrataSelectQuery class, let's move it to be a private helper method in that class.